src/rules.ts

```diff
@@ -63,8 +82,23 @@ export class ResourceGraphRule {
       ruleName: rule.name,
       description: rule.description,
       total: response.totalRecords,
-      ids: resourceIds,
+      resources: resourceIds.map(id => {
+        return {id};
+      }),
     };
     return scanResult;
   }
+
+  private static _queryByResourceGroups(
+    resourceGroups: string[],
+    rule: IResourceGraphRule
+  ) {
+    resourceGroups.forEach((rGroup, i) => {
+      if (i === 0) {
+        rule.query += ` | where resourceGroup=='${rGroup}'`;
```

Comment:
We really don't want to mutate our input parameters like this. We want to modify how we're executing the rule - here we're changing the definition of the rule itself. If we try to run the same rule against a second target, we'll end up with unexpected bugs like zero results because the new query is actually `resources | where resourceGroup=='groupA'| where resourceGroup=='groupB'`